### DISCUSSION TITLE
FAQ: Presenter Questions

### DISCUSSION THREAD

**Discussion by camihmerhar** [githubuniverseworkshops/template-workshop#4] [General]
Howdy Workshop Presenters! 

We wanted to provide you all with a space to ask any and all questions around presenting your workshop at Universe. 

Let us know here and we'll try to get an answer to you and your fellow presenters. 

Cheers,
Your Universe Workshop Team

**Comment by arilivigni** [comment]
Hi team,

In the case of GitHub Actions Importer (GAI) workshop we will require codespaces. 

Current environment for users to run through the lessons for the workshop is captured very well in https://github.com/actions/importer-labs. We are able to create a codespace right in that repository or use that as a template to clone to another organization or personal account of the workshop user. These codespaces are pretty big since we are also planned to spin up containers inside for Jenkins or GitLab. We could also just stay with CI/CD offerings like Azure DevOps  and CircleCI, but that also requires additional setup outside of the GitHub ecosystem and codespaces which may have more challenges.

Current Codespaces for importer-labs is 4-core • 16GB RAM • 32GB

Question should we be cloning this repository into this organization and create it as a template?
Can we use codespaces support for workshop users?

We need these questions answered before moving ahead with the workshop development. We will definitely need an extension on the due date of September 6th.

cc: @ppremk 

PS can we add @sparlant to this organization please.

**Reply by arilivigni** [reply]
@camihmerhar I discussed this with @ppremk and his recommendation is to enable codespaces on this organization (billing should be to GitHub). Also, could I get elevated permissions for the organization level so I can self-serve as the need arises to setup my workshop? If that isn't possible not worries, but I figured I would ask.

**Reply by camihmerhar** [reply]
Howdy @arilivigni , happy to hear we can enable codespaces! Trying myself to get owners permission to this org and will try to get you elevated permissions once I do.

**Reply by arilivigni** [reply]
@camihmerhar I have just a clarifying follow-up question. Will it be ok for customers to `open in a workspace` in this repository, the `githubuniverseworkshops` organization?

**Reply by arilivigni** [reply]
@camihmerhar ☝️

**Reply by arilivigni** [reply]
@camihmerhar, I had a slack conversation with @ppremk and the options in preference order given that we will have a list of customers for the workshop ahead of time and can add them as collaborators.

1. Add customers as collaborators to https://github.com/githubuniverseworkshops/importer-labs repository ahead of the workshop so they can create codepaces in the https://github.com/githubuniverseworkshops/ organization.
2. Let customers know ahead of time that they need an organization that can support the following codespaces requirements
```
  "hostRequirements": {
    "cpus": 4,
    "memory": "8gb",
    "storage": "32gb"
  }
```
Let me know if you have any questions

**Comment by CallMeGreg** [comment]
For the GitHub Advanced Security Certification Workshop, Kristen and I will be using automation to create individual organizations in the [ghas-bootcamp](https://github.com/enterprises/ghas-bootcamp) enterprise for each attendee. This enterprise is used by the field team for partner bootcamps, and has GHAS & Actions covered. The field team is aware that we'll be using it for this purpose, and all organizations will be deleted following Universe via automation.

**Reply by theztefan** [reply]
@rvermeulen @lcartey I think we can use the same organization for the purpose of our workshop as well.

**Comment by robandpdx** [comment]
For the `Understanding the risk of script Injections in GitHub Actions workflows` workshop I will be using actions minutes to demonstrate the workflows with and without script injection vulnerabilities. We will also use GHAS. We will need to have a place for users to fork my workshop repo that has GHAS and actions enabled.

**Reply by robandpdx** [reply]
☝️ 👀 @camihmerhar @a-a-ron @ppremk

**Reply by robandpdx** [reply]
Can we do what @CallMeGreg is doing for his workshop? https://github.com/githubuniverseworkshops/template-workshop/discussions/4#discussioncomment-6885904

**Reply by jenachiccino** [reply]
@ppremk ☝️ could you help answer the question

**Reply by ppremk** [reply]
spinning up multiple orgs may not be recommended. perhaps forking or cloning the repo? let's align during dry run or ping us directly in DM if we've got this understanding wrong @robandpdx

**Comment by abhi-dutta** [comment]
For our session "**Harnessing Copilot: Next-Level Strategies for Advanced Security**" below resources/licenses would be needed:

1) Copilot License for participants
2) Actions Minutes
3) Codespaces enabled on the repo
4) GHAS License enabled

👀 @s-samadi

**Reply by jenachiccino** [reply]
☝️ 👀 @camihmerhar @a-a-ron @ppremk
